src/main.rs

```diff
@@ -63,18 +63,23 @@ impl Snare {
     /// config file. **Note that because snare has multiple threads, the config file can change at
     /// any arbitrary point, not just after calling this function.**
     fn check_for_hup(&self) {
-        if self.sighup_occurred.load(Ordering::Release) {
-            let config = Config::from_path(&self.conf_path);
-            *self.config.lock().unwrap() = config;
-            self.sighup_occurred.store(false, Ordering::Release);
+        if self.sighup_occurred.load(Ordering::Relaxed) {
+            match Config::from_path(&self.conf_path) {
+                Ok(config) => *self.config.lock().unwrap() = config,
+                Err(msg) => eprintln!("{}", msg),
+            }
+            self.sighup_occurred.store(false, Ordering::Relaxed);
         }
     }
 }
 
 /// Exit with a fatal error.
 fn fatal(msg: &str) -> ! {
-    debug_assert!(msg.ends_with('.'));
-    eprintln!("{}.", msg);
+    if msg.ends_with('.') {
+        eprintln!("{}", msg);
+    } else {
+        eprintln!("{}.", msg);
+    }
```

Comment:
I wonder if this is worth it :)

There's a any number of silly things the caller might do.

```
msg("error..");
msg("error,");
...
```

Reply:
Yeah, I'm not sure either :/